src/widgets/section/SectionList.tsx

```diff
@@ -41,12 +48,105 @@ export default class SectionList extends WidgetBase<SectionListParameters> {
 		);
 	}
 
+	private _getContent() {
+		const { section, listIndex } = this.properties;
+		if (listIndex) {
+			return this.meta(Block).run(compiler)(section, listIndex, {
+				locale: i18n.locale
+			})
+		}
+
+		return null;
+	}
+
+	private _parseList(subsection: string, list?: DNode | DNode[]): DNode[] {
+		const { section } = this.properties;
+		if (!list) {
+			return [];
+		}
+
+		if (!Array.isArray(list)) {
+			list = [list];
+		}
+
+		let finalNodes: DNode[] = [];
+		for (let node of list) {
+			if (!node) {
+				continue;
+			}
+
+			if (isVNode(node)) {
+				if (node.tag === 'ul') {
+					node = <ul classes={css.list}>{node.children}</ul>;
+				}
+				else if (node.tag === 'p') {
+					finalNodes.push(...this._parseList(subsection, node.children));
+					continue;
+				}
+				else if (node.tag === 'a') {
+					const match = /.\/([\w\-\._~:\/?[\]@!\$&'\(\)\*\+,;=.]+)([#][\w\-\._~:\/?[\]@!\$&'\(\)\*\+,;=.]+)?$/g.exec(node.properties.href);
+					let page = '';
+					let hash: string | undefined;
+					if (match && match.length >= 2) {
+						page = match[1];
+						if (match.length === 3) {
+							hash = match[2];
+						}
+					}
+					node = (
+						<ActiveHashLink
+							to={`${section}-page`}
+							params={{ subsection, page: page }}
+							hash={hash}
+							classes={css.itemLink}
+							activeClasses={[css.selected]}
+						>
+							{node.children}
+						</ActiveHashLink>
+					);
+				}
+				if (node.children) {
+					node.children = this._parseList(subsection, node.children);
+				}
+			}
+			finalNodes.push(node);
+		}
+
+		return finalNodes;
+	}
+
+	private _onHashChange(event: HashChangeEvent) {
+		this.invalidate();
+	}
+
+	protected onAttach() {
+		window.addEventListener('hashchange', (event) => this._onHashChange(event));
```

Comment:
can use https://github.com/dojo/widgets/tree/master/src/global-event for this inside the render so you can remove the `onAttach` and onDetach`